File: queue.py
def isValid(s):
    stack = []
    for p in s:
        if p == "(":
            stack.append(")")
        elif p == "{"
        stack.append("}")
        elif p == "["
        stack.append("]")
        elif not stack and stack.pop() != p:
            return False
    return not stack

File: pre/queue.js
const s = "{(({[]))[]}";
function queue(s) {
  const stack = [];
  for (let i = 0; i < s.length; i++) {
    if (s == "(") {
      stack.append(")");
    } else if (s == "{") {
      stack.append("}");
    } else if (s == "[") {
      stack.append("]");
    } else if (stack == null || stack.pop() != s) {
      return false;
    } else if (stack == null) {
      return true;
    }
  }
}

queue(s);
